libs/platform/ebpf_object.c

```diff
@@ -368,3 +399,107 @@ ebpf_duplicate_string(_In_z_ const char* source)
     memcpy(destination, source, length);
     return destination;
 }
+
+_Must_inspect_result_ ebpf_result_t
+ebpf_object_acquire_id_reference(ebpf_id_t id, ebpf_object_type_t object_type)
+{
+    ebpf_lock_state_t state = ebpf_lock_lock(&_ebpf_object_tracking_list_lock);
+
+    uint32_t index;
+    ebpf_result_t result = _get_index_from_id(id, &index);
+    if (result != EBPF_SUCCESS) {
+        goto Done;
+    }
+
+    ebpf_id_entry_t* entry = &_ebpf_id_table[index];
+    ebpf_assert(entry->reference_count);
+    if (entry->reference_count == 0) {
+        if (entry->object == NULL) {
+
+            // Do not allow access to entries that are not in use.
+            result = EBPF_INVALID_ARGUMENT;
+        } else {
+
+            // This can _never_ happen.  If the object pointer is not null, the ref-count HAS TO BE _at_ _least_ 1.
+            // This conditon is indicative of a severe internal error.
+            ebpf_assert(0);
+            result = EBPF_FAILED;
+        }
+        goto Done;
+    }
+
+    // ref count is non-zero
+    if (entry->object == NULL) {
+
+        // The object at this entry has been deleted and all that remains are (the now stale) references to this entry
+        // held by other objects.  This is a non-reversible situation and there's no point in giving out references
+        // for such entries, so deny with a suitable error code.
+        result = EBPF_STALE_KEY;
```

Comment:
Shouldn't this be called EBPF_STALE_ID?  Or is there any other type of key this is used for?